packages/ui/src/utils/hooks/useCloseByEscKey.ts

```diff
@@ -0,0 +1,42 @@
+import { useEffect } from 'react'
+
+type CallbackFunction = (e?: KeyboardEvent) => void
+
+// `callback stack` to track callers of useCloseByEscKey
+const _cbStack: CallbackFunction[] = []
+
+function _removeFromStack(cb: CallbackFunction) {
+  const idx = _cbStack.indexOf(cb)
+  if (idx > -1) {
+    _cbStack.splice(idx, 1)
+  }
+}
+
+export default function useCloseByEscKey(cb: CallbackFunction) {
```

Comment:
@utkukaratas 
1. Good thinking with the callback stack, however, wouldn't `stopImmediatePropagation` do the job?
2. We tend to use arrow functions everywhere. Should we do it here as well for consistency?
3. We tend to use named exports instead of default ones. Should we do it here for consistency?
4. We already depend on `react-use`. Should we consider using `useKey` from there?

Reply:
(2) and (3) (4)👍 

(1) `.stopImmediatePropagation` will swallow the event and that might meddle with some other components' event handling. not a conservative choice.

so i'm pushing a version that uses `useKey` underneath. if you think `.stopImmediatePropagation` would be okay then i will get rid of useCloseByEsc module all together and use `useKey` directly in Alert and Toast. up to you guys (cc @aigoncharov @danad02).

Reply:
@utkukaratas solid argument about `stopImmediatePropagation`! Let's try to go with the stack like you suggested.

Reply:
I kind of started thinking about some corner cases. Say you have a dialog (modal) opened. Usually you are able to close it by hitting esc key. What happens if we have a dialog opened, an alert is present and we hit the esc key? Should we even care about these cases?

Reply:
@danad02 good thinking! 
@utkukaratas I suggest we stick with `stopImmediatePropagation` after all. It is simpler and might do more good than bad.

Reply:
We can add an option to **not** stop event propagation if we find it's needed or useful in some kind of scenario.